app/services/replication_transfer_updater.rb

```diff
@@ -0,0 +1,135 @@
+# Copyright (c) 2015 The Regents of the University of Michigan.
+# All Rights Reserved.
+# Licensed according to the terms of the Revised BSD License
+# See LICENSE.md for details.
+
+class ReplicationTransferUpdater
+
+  def self.update(record, params)
+    run_update(build_update(record,params))
+  end
+
+  def self.build_update(record, params)
+    [CancelUpdate, FixityValueUpdate, StoreRequestedUpdate, StoredUpdate].each do |update_class|
+      return update_class.new(record, params) if update_class.matching_update?(record, params)
+    end
+    return ReplicationTransferUpdate.new(record, params)
+  end
+
+  def self.run_update(update)
+    update.update
+  end
+
+  class ReplicationTransferUpdate
+    def initialize(record, params)
+      @record = record
+      @params = params
+    end
+    def update
+      @record.update(@params)
+    end
+    def we_requested?
+      @record.from_node&.namespace == Rails.configuration.local_namespace
+    end
+    def we_replicating?
+      @record.to_node&.namespace == Rails.configuration.local_namespace
+    end
+  end
+
+
+  class CancelUpdate < ReplicationTransferUpdate
+    def self.matching_update?(record, params)
+      record.respond_to?(:cancelled) &&
+        record.cancelled == false &&
+        params[:cancelled] == true
+    end
+    def update
+      @record.cancel!(@params[:cancel_reason])
+    end
```

Comment:
The expected behavior of all the `update` methods is not entirely clear.  The parent class is using only `update` and not `update!`, so it will not raise an exception.  However, this override method is using `cancel!`, so it could raise an exception.  In the class below, it seems to return an update result, so there is some idea of how it should behave.  The behavior of the method through each of the classes that implement it should conform to a consistent, clearly defined role behavior.